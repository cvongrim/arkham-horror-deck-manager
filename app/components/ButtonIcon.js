// React Library Imports
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {StyleSheet, Text, TouchableOpacity} from 'react-native';

// Styles
import COLORS from '../styles/colors';

/**
 * Render a button that has an Icon and Label
 */
class ButtonIcon extends Component {
    /**
     * Constructor
     * @param {object} props
     */
    constructor(props) {
        super(props);
    }

    /**
     * Render the button
     * @return {*}
     */
    render() {
        return (
            <TouchableOpacity
                accessible={true}
                accessibilityLabel={'Open ' + this.props.label}
                style={styles.buttonContainer}
                onPress={() => this.props.action(this.props.link)}>
                {ButtonIcon._renderIcon(this.props.icon, this.props.color)}
                <Text allowFontScaling={false} style={[styles.buttonText]}>{this.props.label}</Text>
            </TouchableOpacity>
        );
    }

    /**
     * Render the icon if available
     *
     * @param {string} icon
     * @param {string} color Color Hex to make icon
     * @return {*}
     * @private
     */
    static _renderIcon(icon = null, color = COLORS.white) {
        if (icon) {
            return (
                <Icon
                    color={color}
                    name={icon}
                    size={30}
                />);
        }
    }
}

ButtonIcon.propTypes = {
    action: PropTypes.func.isRequired,
    color: PropTypes.string,
    icon: PropTypes.string,
    link: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
};

const styles = StyleSheet.create({
    buttonContainer: {
        flex: 1,
        flexDirection: 'row',
        paddingHorizontal: 25,
        paddingVertical: 8,
    },
    buttonText: {
        alignSelf: 'center',
        flex: 1,
        marginLeft: 12,
    },
});

export default ButtonIcon;
