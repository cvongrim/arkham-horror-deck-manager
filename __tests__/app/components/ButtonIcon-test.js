/* global test expect */
/* eslint-disable no-console */
import 'react-native';
import React from 'react';
import ButtonIcon from '../../../app/components/ButtonIcon';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import colorStyles from '../../../app/styles/colors';

test('ButtonIcon renders correctly', () => {
    const tree = renderer.create(<ButtonIcon
        action={(link) => console.log('Tap!')}
        color={colorStyles.grey}
        icon={'menu'}
        key={'menu'}
        label={'Menu'}
        link={'menu'}/>).toJSON();
    expect(tree).toMatchSnapshot();
});
