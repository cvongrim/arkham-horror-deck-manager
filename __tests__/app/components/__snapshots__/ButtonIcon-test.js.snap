// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIcon renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel="Open Menu"
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "opacity": 1,
      "paddingHorizontal": 25,
      "paddingVertical": 8,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#808080",
          "fontSize": 30,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    î—’
  </Text>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "flex": 1,
          "marginLeft": 12,
        },
      ]
    }
  >
    Menu
  </Text>
</View>
`;
